charts/shoot-core/components/charts/kube-proxy/templates/psp/kube-proxy-init-script.yaml

```diff
@@ -0,0 +1,24 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-proxy-init-script
+  namespace: kube-system
+data:
+  init.sh: |
+    #!/bin/sh
+    OLD_KUBE_PROXY_MODE=$(cat /kube-proxy-mode);
+    if [ A${OLD_KUBE_PROXY_MODE} = A${KUBE_PROXY_MODE} ]; then
```

Comment:
Hm, what is this expression? What does it mean? What's the `A`?

Reply:
`sh` does not allow the comparison of an empty strings and will return an error.

Reply:
What about making it more explicit, e.g:

```shell
#!/bin/sh

function foo {
  if [ -z "$1" ] || [ "$1" == "expected-string" ]; then
    echo "exit 0"
  fi
}

foo "$(cat /non-existing-file)"
foo "expected-string"
```

? Both invocations of `foo` return `exit 0`.